context.go

```diff
@@ -1399,6 +1423,57 @@ func (c *Context) addInterVariantDependency(origModule *moduleInfo, tag Dependen
 	atomic.AddUint32(&c.depsModified, 1)
 }
 
+// parentFile returns the path of a file with the same name but in a parent directory
+// For example, if path = "a/b/Blueprints", then parent = "a/Blueprints"
+func parentFile(path string) (parent string, hasParent bool) {
+	dir, leaf := filepath.Split(path)
+	parentDir := filepath.Dir(filepath.Clean(dir))
+	parentFile := filepath.Join(parentDir, leaf)
+	if parentFile == path {
+		return "", false
+	}
+	return parentFile, true
+}
+
+// findBlueprintAncestors returns two maps linking parents to children
+// For example, if paths = []string{"a/b/c/Blueprints", "a/Blueprints"},
+// then ancestors = {"a/b/c/Blueprints":"a/Blueprints"}
+// and descendants = {"a/Blueprints", ["a/b/c/Blueprints"]}
+func findBlueprintAncestors(paths []string) (ancestors map[string]string, descendants map[string][]string) {
```

Comment:
Returning descendants and a list of paths that have no ancestors would be more useful.  Or just return descendants, and put paths with no ancestors as descendants of "".

Reply:
thanks